tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -1269,7 +1270,15 @@ private string GetBranchName()
         }
 
         var gitOutput = GitCommandHelper.RunGitCommand(_workingDirectory, "branch --show-current", MetricTags.CIVisibilityCommands.GetBranch);
-        return gitOutput?.Output.Replace("\n", string.Empty) ?? string.Empty;
+        var res = gitOutput?.Output.Replace("\n", string.Empty) ?? string.Empty;
+
+        if (string.IsNullOrEmpty(res))
+        {
+            Log.Warning("ITR: empty branch indicates a detached head at commit {Commit}", _commitSha);
+            res = $"auto:git-detached-head";
```

Comment:
I assume this is standardised across languages now 🙂

Reply:
We are pioneers, but yep :)